src/vectara/client.py

```diff
@@ -1,10 +1,73 @@
-from .base_client import BaseVectara, AsyncBaseVectara
+import logging
+import threading
+import typing
+
+from datetime import datetime, timedelta
+
+from . import SearchCorporaParameters, GenerationParameters, ChatParameters, ChatFullResponse
+from .base_client import BaseVectara, AsyncBaseVectara, OMIT
 from vectara.managers.corpus import CorpusManager
-from vectara.managers.upload import UploadManager, UploadWrapper
+from vectara.managers.upload import UploadManager
 from vectara.utils import LabHelper
 
-from typing import Union, Optional, Callable
-import logging
+from typing import Union
+
+from .core import RequestOptions
+
+
+class ChatSessionManager:
+    def __init__(self, session_expiry_time: timedelta = timedelta(days=7),
+                 cleanup_interval_in_seconds: int = 43200):
+        self.sessions: typing.Dict[str, typing.Any] = {}
+        self.session_expiry_time = session_expiry_time
+        self.lock = threading.Lock()
+        self.cleanup_event = threading.Event()
+        self.cleanup_interval_in_seconds = cleanup_interval_in_seconds  # Default is set to 12 hours in
+        # seconds
+        self.cleanup_thread = threading.Thread(target=self._run_cleanup, daemon=True)
+        self.cleanup_thread.start()
+
+    def create_session(self, chat_id: str,
+                       search: SearchCorporaParameters,
+                       generation: typing.Optional[GenerationParameters] = OMIT,
+                       chat: typing.Optional[ChatParameters] = OMIT,
+                       request_options: typing.Optional[RequestOptions] = None,
+                       request_timeout: typing.Optional[int] = None,
+                       request_timeout_millis: typing.Optional[int] = None,
+                       ):
```

Comment:
Consider adding return type in this and other functions... Use `None` if nothing is returned